src/compiler/utilitiesPublic.ts

```diff
@@ -366,30 +376,59 @@ export function textSpanOverlap(span1: TextSpan, span2: TextSpan): TextSpan | un
     return overlap && overlap.length === 0 ? undefined : overlap;
 }
 
-export function textSpanIntersectsWithTextSpan(span: TextSpan, other: TextSpan) {
+export function textSpanIntersectsWithTextSpan(span: TextSpan, other: TextSpan): boolean {
     return decodedTextSpanIntersectsWith(span.start, span.length, other.start, other.length);
 }
 
-export function textSpanIntersectsWith(span: TextSpan, start: number, length: number) {
+export function textSpanIntersectsWith(span: TextSpan, start: number, length: number): boolean {
     return decodedTextSpanIntersectsWith(span.start, span.length, start, length);
 }
 
-export function decodedTextSpanIntersectsWith(start1: number, length1: number, start2: number, length2: number) {
+export function decodedTextSpanIntersectsWith(start1: number, length1: number, start2: number, length2: number): boolean {
     const end1 = start1 + length1;
     const end2 = start2 + length2;
     return start2 <= end1 && end2 >= start1;
 }
 
-export function textSpanIntersectsWithPosition(span: TextSpan, position: number) {
+export function textSpanIntersectsWithPosition(span: TextSpan, position: number): boolean {
     return position <= textSpanEnd(span) && position >= span.start;
 }
 
+/** @internal */
+export function textRangeIntersectsWithTextSpan(range: TextRange, span: TextSpan): boolean {
+    return textSpanIntersectsWith(span, range.pos, range.end - range.pos);
+}
+
 export function textSpanIntersection(span1: TextSpan, span2: TextSpan): TextSpan | undefined {
     const start = Math.max(span1.start, span2.start);
     const end = Math.min(textSpanEnd(span1), textSpanEnd(span2));
     return start <= end ? createTextSpanFromBounds(start, end) : undefined;
 }
 
+/** @internal */
+export function normalizeSpans(spans: readonly TextSpan[]): TextSpan[] {

```

Comment:
Can you stick a JSDoc comment describing what "normalization" is, for reference?